dtfs-central-api/src/v1/controllers/portal/facility/delete-amendment.controller.test.ts

```diff
@@ -0,0 +1,110 @@
+
+const facilityId = 'facilityId';
+const amendmentId = 'amendmentId';
+
+const mockDeletePortalFacilityAmendment = jest.fn();
+console.error = jest.fn();
+
+const generateHttpMocks = ({ auditDetails }: { auditDetails: unknown }) =>
+  createMocks<DeletePortalAmendmentRequest>({ params: { facilityId, amendmentId }, body: { auditDetails } });
+
+describe('deletePortalAmendment', () => {
+  beforeEach(() => {
+    jest.resetAllMocks();
+
+    jest.spyOn(PortalFacilityAmendmentService, 'deletePortalFacilityAmendment').mockImplementation(mockDeletePortalFacilityAmendment);
+  });
+
+  it(`should return ${HttpStatusCode.BadRequest} if the audit details are invalid`, async () => {
+    // Arrange
+    const auditDetails = { type: 'not a type' };
+    const { req, res } = generateHttpMocks({ auditDetails });
+
+    // Act
+    await deletePortalAmendment(req, res);
+
+    // Assert
+    expect(res._getStatusCode()).toEqual(HttpStatusCode.BadRequest);
+    expect(res._getData()).toEqual({
+      status: HttpStatusCode.BadRequest,
+      message: "Supplied auditDetails must contain a 'userType' property",
+      code: API_ERROR_CODE.INVALID_AUDIT_DETAILS,
+    });
+  });
+
+  it('should call PortalFacilityAmendmentService.deletePortalFacilityAmendment with the correct params', async () => {
+    // Arrange
+    const auditDetails = generatePortalAuditDetails(aPortalUser()._id);
+    const { req, res } = generateHttpMocks({ auditDetails });
+
+    // Act
+    await deletePortalAmendment(req, res);
+
+    // Assert
+
+    expect(mockDeletePortalFacilityAmendment).toHaveBeenCalledTimes(1);
+    expect(mockDeletePortalFacilityAmendment).toHaveBeenCalledWith({
+      amendmentId,
+      facilityId,
+      auditDetails,
+    });
+  });
+
+  it(`should return ${HttpStatusCode.NoContent}`, async () => {
+    // Arrange
+    const auditDetails = generatePortalAuditDetails(aPortalUser()._id);
+    const { req, res } = generateHttpMocks({ auditDetails });
+
+    // Act
+    await deletePortalAmendment(req, res);
+
+    // Assert
+    expect(res._getStatusCode()).toEqual(HttpStatusCode.NoContent);
+  });
+
+  it('should return the correct status and body if PortalFacilityAmendmentService.deletePortalFacilityAmendment throws an api error', async () => {
+    // Arrange
+    const status = HttpStatusCode.Forbidden;
+    const message = 'Test error message';
+    mockDeletePortalFacilityAmendment.mockRejectedValue(new TestApiError({ status, message }));
+
+    const auditDetails = generatePortalAuditDetails(aPortalUser()._id);
+    const { req, res } = generateHttpMocks({ auditDetails });
+
+    // Act
+    await deletePortalAmendment(req, res);
+
+    // Assert
+    expect(res._getStatusCode()).toEqual(status);
+    expect(res._getData()).toEqual({
+      status,
+      message,
+    });
+  });
+
+  it(`should return ${HttpStatusCode.InternalServerError} if PortalFacilityAmendmentService.updatePortalFacilityAmendment throws an unknown error`, async () => {
```

Comment:
Can you please add an assertion with `facilityId` and `amendmentId` to be `invalid` to validate Mongo ID check.